### PR TITLE
Trev8939/feature rend mode dev

### PR DESCRIPTION
**Pull request by TADraeseke**
Adds sample demonstrating differences between static and dynamic rendering of feature layers in a scene view.

Also updates naming in the similar sample (for maps instead of scene views), for clarity.

### CODE REVIEW

**Comment by TADraeseke** [comment]
@doneill I know this may be late in the game. If you think you don't have time for the review, just let me know!

File: java/feature-layer-rendering-mode-scene/build.gradle [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+
+    defaultConfig {
+        applicationId "com.esri.arcgisruntime.sample.featurelayerrenderingmodescene"
+        minSdkVersion 22
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        versionCode 1
+        versionName "1.0"
```
Comment (doneill) [inline java/feature-layer-rendering-mode-scene/build.gradle]: `versionCode` and `versionName` should come from global vars

```groovy
versionCode rootProject.ext.versionCode
versionName rootProject.ext.versionName
```

This should be updated in the rendering mode map sample as well.

File: java/feature-layer-rendering-mode-scene/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            android:text="@string/renderingmode_dynamic"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:id="@+id/dynamicTextView"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:layout_marginBottom="32dp"
+            android:layout_marginStart="8dp"
+            app:layout_constraintLeft_toLeftOf="parent"
+            android:textStyle="bold"
+            android:textColor="@android:color/background_light"/>
```
Comment (doneill) [inline java/feature-layer-rendering-mode-scene/src/main/res/layout/activity_main.xml]: Better seen with `background_dark`, same with TextView above.

File: java/feature-layer-rendering-mode-scene/src/main/java/com/esri/arcgisruntime/sample/featurelayerrenderingmodescene/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        getResources().getString(R.string.energy_geology_feature_service) + "0");
+    ServiceFeatureTable contactsServiceFeatureTable = new ServiceFeatureTable(
+        getResources().getString(R.string.energy_geology_feature_service) + "8");
+    ServiceFeatureTable outcropServiceFeatureTable = new ServiceFeatureTable(
+        getResources().getString(R.string.energy_geology_feature_service) + "9");
+
+    // create the feature layer using the service feature table
+    FeatureLayer faultFeatureLayer = new FeatureLayer(faultServiceFeatureTable);
+    FeatureLayer contactsFeatureLayer = new FeatureLayer(contactsServiceFeatureTable);
+    FeatureLayer outcropFeatureLayer = new FeatureLayer(outcropServiceFeatureTable);
```
Comment (doneill) [inline java/feature-layer-rendering-mode-scene/src/main/java/com/esri/arcgisruntime/sample/featurelayerrenderingmodescene/MainActivity.java]: I think you should check the load status of these layers, they do not always render for me.  

Here is an example where the top layers load: 

notrendering

Here nothing loads

lesrendering

**Comment by TADraeseke** [comment]
Thanks @doneill!

There are still a few bugs with the feature layer work and those layers sometimes don't draw for me either (though they do for the last 2 daily builds on the ZenFone).

I reproduced the not drawing on older builds, but unfortunately they're still reported as `LoadStatus.LOADED` even when not visible on those previous builds.

**Comment by doneill** [comment]
@TADraeseke I updated the daily build and retested on Nexus 6 where the sample worked as expected.

**Pull request merged**